### PR TITLE
Issue #13100: Create tests from regexp category checks xdoc examples

### PR DESCRIPTION
**Pull request by stoyanK7**
Part of #13100

https://checkstyle.org/checks/regexp/index.html

### CODE REVIEW

**Comment by stoyanK7** [comment]
Github, generate site

**Comment by romani** [comment]
@stoyanK7 , conflict

File: src/main/java/com/puppycrawl/tools/checkstyle/site/ExampleMacro.java [hunk 1]
```diff
@@ -113,20 +111,33 @@ private static List<String> readFile(String path) throws MacroExecutionException
...
      */
-    private static String getSnippetBetweenDelimiters(
-            Collection<String> lines, String startingDelimiter, String endingDelimiter) {
+    private static String getConfigSnippet(Collection<String> lines) {
         return lines.stream()
-                .dropWhile(line -> !line.contains(startingDelimiter))
+                .dropWhile(line -> !XML_CONFIG_START.equals(line))
                 .skip(1)
-                .takeWhile(line -> !line.contains(endingDelimiter))
+                .takeWhile(line -> !XML_CONFIG_END.equals(line))
```
Comment (stoyanK7) [inline src/main/java/com/puppycrawl/tools/checkstyle/site/ExampleMacro.java:125]: Explanation for changes. `contains()` is not good enough for the ending config delimeter - `*/`. There are configs that contain these two characters and the config is cutoff before the end.

For example this 
```xml
        <source>
&lt;module name="Regexp"&gt;
  &lt;property
    name="format"
    value="\A/{71}\n// checkstyle:\n// Checks Java source code for
    adherence to a set of rules\.\n// Copyright \(C\) \d\d\d\d  Oliver Burn\n
    // Last modification by \$Author.*\$\n/{71}\n\npackage [\w\.]*;\n\n
    (import [\w\.]*;\n)*\n/\*\*\n( \*[^/]*\n)* \*/"/&gt;
&lt;/module&gt;
        </source>
```

would produce the following output
```xml
&lt;module name="Regexp"&gt;
  &lt;property
    name="format"
    value="\A/{71}\n// checkstyle:\n// Checks Java source code for
    adherence to a set of rules\.\n// Copyright \(C\) \d\d\d\d  Oliver Burn\n
    // Last modification by \$Author.*\$\n/{71}\n\npackage [\w\.]*;\n\n
    (import [\w\.]*;\n)*\n/\*\*\n( \*[^/]*\n)* \*/"/&gt;
```

**Review by romani — APPROVED** [review]
ok to merge

File: src/main/java/com/puppycrawl/tools/checkstyle/site/ExampleMacro.java (hunk 1 above)
Comment (nrmancuso) [reply src/main/java/com/puppycrawl/tools/checkstyle/site/ExampleMacro.java:125]: Better to be specific.

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**